web-registry/src/lib/wallet/wallet.utils.ts

```diff
@@ -0,0 +1,85 @@
+import { MutableRefObject } from 'react';
+import WalletConnect from '@walletconnect/client';
+import truncate from 'lodash/truncate';
+
+import { UseStateSetter } from 'types/react/use-state';
+
+import { chainInfo } from './chainInfo/chainInfo';
+import { Wallet } from './wallet';
+import { walletConnectClientMeta } from './wallet.constants';
+import {
+  WalletClient,
+  WalletConfig,
+} from './walletsConfig/walletsConfig.types';
+
+/* getWalletConnectInstance */
+
+type GetWalletConnectInstanceParams = {
+  setWalletConnectUri: UseStateSetter<string | undefined>;
+  onQrCloseCallback: MutableRefObject<(() => void) | undefined>;
+};
+
+export const getWalletConnectInstance = async ({
+  setWalletConnectUri,
+  onQrCloseCallback,
+}: GetWalletConnectInstanceParams): Promise<WalletConnect> => {
+  const walletConnect = new (await import('@walletconnect/client')).default({
+    bridge: 'https://bridge.walletconnect.org',
+    signingMethods: [
+      'keplr_enable_wallet_connect_v1',
+      'keplr_sign_amino_wallet_connect_v1',
+    ],
+    qrcodeModal: {
+      open: (uri: string, callback: () => void) => {
+        // Open QR modal by setting URI.
+        setWalletConnectUri(uri);
```

Comment:
could you explain a bit more about what this URI is and how it is working? this might be a good spot in the code to some comments documenting this a little bit.

Reply:
also curious about what this `callback` does, it seems like it later becomes the `onQrCloseCallback`. maybe a rename here would be helpful for consistency?

Reply:
Good point, I added the related explanation from their doc here:
```
// The protocol establishes a remote connection between two apps and/or devices using a Bridge server to relay payloads.
// These payloads are symmetrically encrypted through a shared key between the two peers.
// The connection is initiated by one peer displaying a QR Code or deep link with a standard WalletConnect URI and is established when the counter-party approves this connection request.
```

Reply:
`onQrCloseCallback ` is a callback to clean the URI within wallet connect once the connection has been established.
Agree with the naming, will make it more explicit.